cdap-api-common/src/main/java/io/cdap/cdap/api/common/HttpUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.common;
+
+import java.net.HttpURLConnection;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Utility functions for HTTP processing.
+ */
+public final class HttpUtils {
```

Comment:
does this need to be in cdap-api-common? It seems like it's only used internally?

Also think HttpCodes would be a slightly better name, we usually try to make util classes very specific in scope, otherwise they tend to accumulate a bunch of unrelated logic.

Reply:
Will rename to HttpCodes.

It's used in cdap-app-fabric, cdap-common and cdap-etl. Is there a better 'common' module to move this class?

Reply:
I see, there isn't one today, as cdap-etl is an application module which shouldn't depend on cdap internals. In this particular case, I don't think it's super important to try and share the exact same constant set in the platform and in the app. Can we just define them separately?

The issue with putting it in API is in maintaining it. For example, if it had already been there, we would probably not have been able to add a new http code to the set, as that could would be a change in API behavior.

Reply:
I have duplicated the `HttpCodes` class for now (removed the class cdap-api-common) as would like to fix the retry issue in 6.8.1. Created https://cdap.atlassian.net/browse/CDAP-20266 to fix this properly in develop (6.9.0) by creating a commons module like you suggested.